### PR TITLE
Ensure PT model is in evaluation mode and lightweight forward pass done

### PR DESCRIPTION
**Pull request by amyeroberts**
Small update to the `pt-to-tf` CLI. Sets the pytorch model into evaluate model and uses `no_grad` context to make the memory requirements lighter. 

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

### CODE REVIEW

File: src/transformers/commands/pt_to_tf.py [hunk 1]
```diff
@@ -270,9 +270,13 @@ def run(self):
 
         # Load models and acquire a basic input compatible with the model.
         pt_model = pt_class.from_pretrained(self._local_dir)
+        pt_model.eval()
```
Comment (sgugger) [inline src/transformers/commands/pt_to_tf.py:273]: If you don't get a model in `eval` mode with `from_pretrained`, then it's a bug.
(Not saying we shouldn't be safe, but this line shouldn't be necessary.)

**Review by sgugger — APPROVED** [review]
Thanks!

**Review by gante — APPROVED** [review]
TIL :D

(LGTM 👍 )
